dex/src/main/scala/com/wavesplatform/dex/api/websockets/actors/PingPongHandlerActor.scala

```diff
@@ -0,0 +1,100 @@
+package com.wavesplatform.dex.api.websockets.actors
+
+import java.util.UUID
+
+import akka.actor.{Actor, Cancellable, Props}
+import akka.http.scaladsl.model.ws.TextMessage
+import cats.syntax.option._
+import com.wavesplatform.dex.api.MatcherWebSocketRoute.ConnectionSource
+import com.wavesplatform.dex.api.websockets.WsMessage
+import com.wavesplatform.dex.api.websockets.actors.PingPongHandlerActor._
+import com.wavesplatform.dex.api.websockets.statuses.TerminationStatus
+import com.wavesplatform.dex.api.websockets.statuses.TerminationStatus._
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+import scala.concurrent.duration._
+
+class PingPongHandlerActor(settings: Settings, maxConnectionLifetime: FiniteDuration, connectionSource: ConnectionSource)
+    extends Actor
+    with ScorexLogging {
+
+  import context.dispatcher
+
+  private val maxLifetimeExceeded = scheduleOnce(maxConnectionLifetime, CloseConnection(MaxLifetimeExceeded(connectionSource.id)))
+  private val firstPing           = scheduleOnce(settings.pingInterval, SendPing)
+
+  private def scheduleOnce(delay: FiniteDuration, message: Any): Cancellable = context.system.scheduler.scheduleOnce(delay, self, message)
+
+  private def awaitPong(expectedPong: PingOrPong, scheduledTimeouts: List[Cancellable], nextPing: Cancellable): Receive = {
+
+    case SendPing => context.become { sendPingAndScheduleNextOne(scheduledTimeouts) }
+
+    case TextMessage.Strict(maybePong) =>
+      Json
```

Comment:
Found an interesting case: a client can send multiple same pongs. Nothing happens except the first pong. But we can not do an unnecessary work

Reply:
Fixed